### DISCUSSION TITLE
Uploading files to Budibase external datasource and referencing them

### DISCUSSION THREAD

**Discussion by IvanDrag0** [Budibase/budibase#14163] [Help]
Budibase has an attachment [column](https://docs.budibase.com/docs/attachments) and [component](https://docs.budibase.com/docs/attachment), however, it looks like it can only be used if the user uses the Budibase DB (instead of an external SQL database).

- Is it possible to create a file upload/attachment column/component using an external SQL database?
- If not, how would I go about adding file attachment/upload capabilities to my application if I'm only using an SQL database? Would I need to use an SQL database and then the Budibase DB for the attachments? If so, how would I reference/link them together?

**Comment by ConorWebb96** [comment]
Hey @IvanDrag0,

I believe there already is a custom plugin that does exactly this on our [plugins repository](https://github.com/Budibase/plugins).  I believe it is called `Budibase file upload`.

**Reply by egutierrez-ar** [reply]
Not the original author, but I also face this issue.

I don't believe this plugin is a suitable replacement to the requirement, as there's many caveats with uploading the content of a file to a transactional database, let alone multiple files for a single record.

I think this plugin has a _slightly_ better idea on what the end result should be (development experience wise), however it's made with some major hacks that make the url of the uploaded document expire aswell as require elevated admin permissions to operate.

https://github.com/mrkking/budibase-object-store-field

Is there any plan to release some sort of feature to allow for such flexibility to store files in block storage (s3 or like), while keeping the references in an external database?
